fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -384,18 +401,30 @@ def state_dict(self, *args, **kwargs):  # type: ignore
         .. warning:: This needs to be called on all ranks, since synchronization
             primitives will be used.
         """
-        with self.summon_full_params():
-            # Buffers dtype stays consistent with parameters.
-            self._all_buffers_to(dtype=torch.float32)
-
-            state_dict = self.module.state_dict(*args, **kwargs)
-            # We copy the state_dict since full param will be freed after
-            # we exit the summon_full_params() context.
-            for key in state_dict.keys():
-                state_dict[key] = state_dict[key].clone()
+        # TODO
+        # Buffers dtype stays consistent with parameters.
+        # self._all_buffers_to(dtype=torch.float32)
```

Comment:
Removing these doesn't seem to cause any tests failures. It's not clear to me if this should be removed? Or if we should add more tests?

Reply:
more likely I think we just don't have the test. I will file an issue.

Reply:
or do you wanna address it here? I am fine either way.

Reply:
I see, we don't maintain the invariant that buffers are cast back to fp32 at the end of each `forward()`. Actually, I'm realizing we don't maintain fp32 buffers at all, so there could be loss of precision throughout training... we should fix that, perhaps in a separate PR: https://github.com/facebookresearch/fairscale/issues/442

Reply:
self-assigned #442. I wrote the existing buffer logic with the incorrect assumption that buffers are static.